### DISCUSSION TITLE
Feature Request: Podman support

### DISCUSSION THREAD

**Discussion by kholbekj** [basecamp/kamal#1073] [Ideas]
Podman has a number of benefits on the development machine over docker. It would be great if mrsk would leverage either.

I don't think the delta between the current build+push process with docker and podman is huge, the registry would have to always be supplied explicitly rather than rely on dockers defaulting, and the `buildx build` command behaves a little differently. the `--builder` param doesn't exist, and `--push` neither, so breaking into a few steps might be needed.

I'm not really a very proficient docker user, so I'm a bit unsure how to tackle the `--builder` thing, I assume the push issue can be resolved by just pushing in a second command.

**Comment by jeremy** [comment]
Good notes. Duplicates #61.

**Comment by kholbekj** [comment]
Whops, my bad!

**Comment by igrigorik** [comment]
#61 was resolved without action. Can we reopen this? Would love to see podman support as alternative to Docker.

**Comment by gabrielperales** [comment]
Any update on this? 👀

**Comment by jeremy** [comment]
Please - do investigate

**Comment by phoozle** [comment]
I am currently working on a Podman proof of concept if anyone is interested. Give me a shout if you want to lend a hand or help me test.
https://github.com/phoozle/kamal_podman/

There is a feature matrix on the project to keep track of what functionality is/isn't implemented or not supported with Podman.

**Reply by gabrielperales** [reply]
Hi Matthew, I don't know how I can help with the development, but maybe I can help you testing it :-)

**Reply by viktorianer** [reply]
Me 2. Just do not have time for development.

I gave up on Podman, when I tried to run it with Kamal.

<img width="942" alt="Screenshot 2024-10-17 at 18 12 20" src="https://github.com/user-attachments/assets/1659e082-d86c-4cd0-997a-cd8d00cfddf6">

**Reply by phoozle** [reply]
Thanks for trying it out.

Let's move the discussion for my branch over to here:
https://github.com/basecamp/kamal/discussions/1256

I would suggest pulling down my branch again and make sure you're on the latest commit and try running `kamal build push`

Currently I only have support for local build with podman.
